dask/dataframe/tests/test_dataframe.py

```diff
@@ -2189,6 +2189,27 @@ def test_fillna():
     assert_eq(df.fillna(method="pad", limit=3), ddf.fillna(method="pad", limit=3))
 
 
+def test_from_delayed_lazy_if_meta_provided():
+    """Ensure that the graph is 100% lazily evaluted if meta is provided"""
+
+    @dask.delayed
+    def raise_exception():
+        raise RuntimeError()
+
+    tasks = [raise_exception()]
+    ddf = dd.from_delayed(tasks, meta=dict(a=float))
+
+    with pytest.raises(RuntimeError):
+        ddf.compute()
+
+
+def test_from_delayed_empty_meta_provided():
+    ddf = dd.from_delayed([], meta=dict(a=float))
+    df = ddf.compute()
+    expected = pd.DataFrame({"a": [0.1]}).iloc[:0]
+    assert_eq(df, expected)
```

Comment:
It's better not to let `assert_eq` call `compute` internally

```suggestion
    expected = pd.DataFrame({"a": [0.1]}).iloc[:0]
    assert_eq(ddf, expected)
```

Reply:
But you are proposing to let it be called internally, don't you? Did you mean "**It is better** to let assert_eq call compute"?

Reply:
Yes :facepalm: "**It is better** to let assert_eq call compute" is what I mean.